### DISCUSSION TITLE
Whats the difference between 'assertEq' and 'assert'

### DISCUSSION THREAD

**Discussion by owanemi** [Cyfrin/foundry-full-course-cu#2141] [Q&A] [answered]
```
        assertEq(address(fundMe).balance, 0);
        assertEq(startingFundMeBalance + startingOwnerBalance, fundMe.getOwner().balance);
```
in the test section of fund me, when patrick writes a test for checking if withdrawal is possible from mutiple owners, he used 'assert' instead of 'assertEq' chatgpt said the difference really doesnt matter because they do same thing, but i'd like to inquire from other cyfrin users.

**Comment by EngrPips** [comment, marked as answer]
Hello @owanemi, ChatGPT is kind of correct that they might be doing the same thing, but not in all cases. I will say `assert` has a broader scope than `assertEq`, the reason being `assert` checks to confirm that any logic you pass into it is true. If not it will error. Let's look at the example below 

**ASSERT EXAMPLE**

```js
uint256 number = 1;
uint256 myNumber = 1
assert(number == myNumber);
assert(number >=  myNumber);
assert(number <= myNumber);
```
You see how I used `assert` on different comparison logic, but this can't be the case for `assertEq` as it is more focused on confirming that a value equals another value. 

**ASSERT EQUAL EXAMPLE**

```js
uint256 number = 1;
uint256 myNumber = 1
assertEq(number, myNumber);
```
You see above that when I used the `assertEq` function, I didn't use any logic inside of it, but instead, I passed in the two values I wanted to compare. To cut the long story short, `asssert` is for general purposes as you pass in the logic for the check you want to do, while `assert` has a specific purpose of checking and confirming that two values are the same

**Reply by owanemi** [reply]
thank you so much!

**Reply by EngrPips** [reply]
You are welcome, buddy.

**Reply by yawarasuuna** [reply]
That's clarifying, thank you for the nuanced response. Sometimes I tend to miss these subtle differences, hopefully it improves as we progress with our coding skills 😂

**Reply by EngrPips** [reply]
Yeah, we only get better with time, If we keep working hard.

**Comment by WillStansill** [comment]
My main understanding of it is that assertEq() takes two variables and makes sure they are equal. So for example on assert you could have something like 

’ assert(!ContractIsBroken) ‘
Which is saying assert contract is broken is not true.

While assertEq taking two variables would more be like this 
‘
A = 5
B = 5
assertEq(A, B)
‘
This use of assertEq takes the two variables defined above and makes sure they are equal to each other.


The reason ChatGPT says they are the same thing is you’ll often find people saying things like 
‘’’
assert(A == B)
‘’’
Which in that case is basically the same thing

**Reply by EngrPips** [reply]
Indeed!
